Gems/LmbrCentral/Code/Tests/Builders/LuaBuilderTests.cpp

```diff
@@ -62,6 +62,59 @@ namespace UnitTest
         AZ::ComponentApplication::Descriptor m_descriptor;
     };
 
+    TEST_F(LuaBuilderTests, ParseLuaScript_DependencyRegexStress_Success)
+    {
+        LuaBuilderWorker worker;
+
+        AssetBuilderSDK::ProductPathDependencySet pathDependencies;
+
+        char resolvedPath[AZ_MAX_PATH_LEN];
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath("@gemroot:LmbrCentral@/Code/Tests/Lua/dependencyRegexStress.lua", resolvedPath, AZ_MAX_PATH_LEN);
+
+        worker.ParseDependencies(AZStd::string(resolvedPath), pathDependencies);
```

Comment:
This could use `resolvedPath.String()` if the type changes to AZ::IO::FixedMaxPath